ios-xcframework/Podfile

```
@@ -9,57 +10,42 @@ HERMES_ENABLED = ENV.fetch('HERMES_ENABLED', true)
 
 puts "[Gutenberg] Installing pods with Hermes #{HERMES_ENABLED ? 'enabled' : 'disabled'}"
 
-# Note that the pods in this array might seem unused if you look for
-# `import` statements in this codebase. However, make sure to also check
-# whether they are used in the gutenberg-mobile and Gutenberg projects.
-#
-# Also notice that these are not all the dependencies the project uses.
-# Later in the config, we call use_react_native! which fetches more.
-#
-# See https://github.com/wordpress-mobile/gutenberg-mobile/issues/5025
-DEPENDENCIES = %w[
-  react-native-safe-area
-  react-native-safe-area-context
-  react-native-video
-  react-native-webview
-  RNSVG
-  react-native-slider
-  BVLinearGradient
-  react-native-get-random-values
-  react-native-blur
-  RNScreens
-  RNReanimated
-  RNGestureHandler
-  RNCMaskedView
-  RNCClipboard
-  RNFastImage
-  React-jsc
-].freeze
-
-def gutenberg_dependencies
-  podspec_prefix = '..'
-
-  computed_dependencies = DEPENDENCIES.dup
-
-  # Use a custom RNReanimated version while we coordinate a fix upstream
-  computed_dependencies.delete('RNReanimated')
-
-  computed_dependencies.delete('React-jsc') unless HERMES_ENABLED
-
-  computed_dependencies.each do |pod_name|
-    pod pod_name, podspec: "#{podspec_prefix}/third-party-podspecs/#{pod_name}.podspec.json"
-  end
-
+def setup_rnreanimated_pod_from_fork
   # This is required to workaround an issue with RNReanimated after upgrading to version 2.17.0
   rn_node_modules = File.join(Dir.pwd, '..', 'gutenberg', 'node_modules')
   raise "Could not find node modules at given path #{rn_node_modules}" unless File.exist? rn_node_modules
+
   ENV['REACT_NATIVE_NODE_MODULES_DIR'] = rn_node_modules
   puts "[Gutenberg] Set REACT_NATIVE_NODE_MODULES_DIR env var for RNReanimated to #{rn_node_modules}"
 
   # Use a custom RNReanimated version while we coordinate a fix upstream
   pod 'RNReanimated', git: 'https://github.com/wordpress-mobile/react-native-reanimated', branch: 'wp-fork-2.17.0'
 end
 
+def gutenberg_dependencies
+  podspec_extension = '.podspec.json'
+
+  Dir["../third-party-podspecs/*#{podspec_extension}"].each do |podspec_path|
+    pod_name = JSON.parse(File.read(podspec_path))['name']
+
+    raise "Could not find pod name in JSON spec at #{podspec_path}" unless pod_name
+
+    # Skip the standard RNReanimated podspec because we use a custom version
+    # while working on an upstream fix
+    #
+    # https://github.com/software-mansion/react-native-reanimated/pull/4684
+    if pod_name == 'RNReanimated'
+      setup_rnreanimated_pod_from_fork
+      next
+    end
+
+    # Skip React-jsc unless using Hermes
+    next if pod_name == 'React-jsc' && HERMES_ENABLED == false
```

When Hermes is enabled we'd only need to include `React-hermes`, so in theory, we could filter out `React-jsc`. I know the condition was set this way but I'm wondering if it was originally wrong 🤔.

---

I've generated the XCFramework without the `React-jsc` pod and the build succeeded locally. I've also tested using a locally generated XCFramework in the WordPress-iOS project and the app runs successfully too. That said, I think we could safely remove it.

```suggestion
    next if pod_name == 'React-jsc' && HERMES_ENABLED == true
```

---

Good point @fluiddot. 

I must have gotten mixed up when changing the syntax and didn't make the JSC = JavaScriptCore connection.

I guess we never noticed because we've been only testing with Hermes enabled for a while. 

I decided not to use your suggestion because I wanted to add more details, see ad3c8f9098b90d8ff4677f33efb3fd1c6abcfbb7